extensions/amp-story-captions/0.1/amp-story-captions.js

```diff
@@ -27,8 +36,22 @@ export class AmpStoryCaptions extends AMP.BaseElement {
 
   /** @override */
   buildCallback() {
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.element.appendChild(this.container_);
+    this.container_ = <div />;
+    this.checkAndMaybeSetPreset_();
+  }
+  /**
+   * Checks if data preset is defined and valid.
+   * If valid it renders in shadow dom with preset syles.
+   * @private
+   */
+  checkAndMaybeSetPreset_() {
```

Comment:
This method actually appends the container depending on how the preset is configured, so the name is not very accurate. I actually suggest having this logic directly in `buildCallback` since creating the `container_` in the same place that it's appended makes the most sense (currently if you want to understand this function you need to also look at buildCallback anyways to know what the container is), and this method doesn't have complex logic so the `buildCallback` won't be cluttered with these 7 lines.